docs/reference/migrating-example.md

```diff
@@ -0,0 +1,171 @@
+
+This migration allowed us to further explore this process and understand typical stumbling blocks and areas that need improvement.
+
+This document outlines that experience and our findings, and provides a case study that should approximate a real-world migration.
+You can see the results by comparing [the initial state of the repo](https://github.com/gradle/gradle-client/tree/4fb821caecbc4a55c5896624b31b201f66cc9fd0) with the state [after the migration was completed](https://github.com/gradle/gradle-client/commit/0c324821e52e82a14413f0d195ad3dda8dc687df).
+
+# How to Convert an Existing Project to Declarative Gradle
+
+The first thing to keep in mind is that using Declarative Gradle is *not* an all or nothing proposition.
+Gradle has always supported mixing build script DSLs between projects in a multi-project build using Groovy and Kotlin DSLs.
+It continues to support doing the same with DCL buildscripts, which allows us to migrate project by project.
+
+# Migration Process for Gradle-Client
+
+## Initial Setup
+
+1. Ensure the latest Gradle wrapper version
+
+Features are constantly being added to the Declarative Gradle support.  
+Any project interested in experimenting with DCL files should ensure they are using at least the latest stable Gradle release.
+For the migration of the Gradle-Client project, we upgraded the wrapper to use a nightly Gradle build [gradle-8.14-20250222002553+0000](https\://services.gradle.org/distributions-snapshots/gradle-8.14-20250222002553+0000-bin.zip) in order to get support for the very latest Declarative Gradle features.
+It is likely that by the time you are reading this, there will be a later milestone or release candidate of Gradle 8.14 available that would be usable for your own migrations.
+
+3. Identify the project to be migrated first
+
+We decided to start with the [`gradle-client` subproject](https://github.com/gradle/gradle-client/tree/main/gradle-client), as this was the most important and fully featured project contained in the Gradle Client build, which would be the most interesting to investigate.
+For your own migrations, you should probably consider which projects would be _simplest_ to migrate.
+Projects that are simpler and most similar to our declarative samples are good candidates.
+If your project _doesn't_ already use Gradle's Kotlin DSL, you'll want to migrate to this first from your Groovy build using the existing [migration guide](https://docs.gradle.org/current/userguide/migrating_from_groovy_to_kotlin_dsl.html) in the Gradle documentation.
+
+3. Add the proper ecosystem plugin
+
+The Declarative Gradle prototype maintains a KMP prototype plugin.
+As the `gradle-client` project is a Kotlin Multiplatform project, we added this plugin to the project's existing `settings.gradle.kts` file.
+Also, be sure to include the `gradlePluginPortal()` as a repository if your project doesn't already include this.
+
+
+```
+pluginManagement {
+    repositories {
+        gradlePluginPortal()
```

Comment:
This is the default?

Reply:
What do you mean?  Removing this from the included `declarative-logic` build in the Gradle Client breaks that build.

Reply:
In this place it is the `settings` file for the outer build, me thinks. Then it's not necessary here,

💭 It would be good to have the path of the file for each snippet to make navigating the snippets easier.